### PR TITLE
[resources/logging] log args as JSON, log message on errors

### PR DESCRIPTION
**Pull request by pawelmhm**
follow up on #274 args message should be proper JSON, commit also adds logging of parsed url representation so that it should be easy to extract netloc and other url components for example using logstash.

### CODE REVIEW

File: splash/resources.py [hunk 1]
```diff
@@ -201,9 +201,27 @@ def _log_stats(self, request, options):
...
+    def detailed_options(self, options):
+        url = options.get("url")
+        if not url:
+            return options
+        url = six.moves.urllib.parse.urlparse(url)
+        options.update({
+            "netloc": url.netloc,
+            "query": url.query,
+            "path": url.path
+        })
```
Comment (kmike) [inline splash/resources.py]: I think we shouldn't parse URLs here - Splash script users may pass anything in 'url' argument, it doesn't have a special meaning. Is itmust be possible to write these filters in Kibana instead?

Comment (pawelmhm) [inline splash/resources.py]: actually parsing url in kibana could be tricky, i can do some research though. Alternative apprach would be: If you think users could pass anything maybe we could just do some try - except, and try parsing url only if it is valid url?

Comment (redapple) [inline splash/resources.py]: I agree with @kmike , `url` could be anything. I would rather do the extra parsing outside of Splash, in log analyzers, like Logstash for example, which can be extended to do URL parsing, or may have that already.

I believe leaving this PR's purpose of making parsing of arguments easier as proper JSON is already a big win, without analyzing them fully at this stage.

Comment (pawelmhm) [inline splash/resources.py]: @redapple @kmike ok I updated PR

**Comment by pawelmhm** [comment]
hey @kmike do you think we should also log Splash status code and client ip in "events"?

**Comment by pawelmhm** [comment]
6e20f66eba93f63888ea573a74b adds proper JSON logging of errors

**Comment by pawelmhm** [comment]
ping @kmike do you think its ok?

File: splash/resources.py [hunk 2]
```diff
@@ -201,9 +203,19 @@ def _log_stats(self, request, options):
...
         }
-        log.msg(json.dumps(msg), system="events")
+        if error:
+            msg["error"] = error
+        try:
+            msg = json.dumps(msg)
+        except TypeError:
+            # splash-as-proxy options may contain byte objects which are not
+            # serializable
+            pass
```
Comment (kmike) [inline splash/resources.py]: splash-as-a-proxy feature is dropped - is this still required?

**Comment by kmike** [comment]
Yeah, I think this feature is good :+1:

File: splash/resources.py (hunk 2 above)
Comment (pawelmhm) [inline splash/resources.py]: good point will check that and update PR

**Comment by pawelmhm** [comment]
rebased against recent master (without splash-as-proxy) and ammended recent commit to log error in consistent format (same format that goes to response will be logged) and without ugly try-except

**Pull request merged**